docs/build/build-with-ai/build-with-goat/send-token-agent.md

```diff
@@ -0,0 +1,228 @@
+
+**3. Getting On-Chain Tools (Token Sending Focus):**
+
+```javascript
+// 2. Get your onchain tools for your wallet
+const tools = await getOnChainTools({
+  wallet: viem(walletClient),
+  plugins: [
+    sendETH(), // Enable ETH transfers - Adapt for CELO
+    erc20({ tokens: [USDC, PEPE] }), // Enable ERC20 token operations - Review for Celo tokens
+  ],
+});
+```
+
+- **On-Chain Tools for Token Sending:** This section configures the tools for sending tokens:
+  - `sendETH()`: Enables sending native ETH on Sepolia. We will adapt this to send native CELO on Celo.
+  - `erc20({ tokens: [USDC, PEPE] })`: Enables ERC20 token operations for USDC and PEPE. We will need to review and potentially replace these tokens with relevant ERC20 tokens on Celo.
+
+**4. Command Line Interface and AI Interaction Loop:**
+
+```javascript
+// 3. Create a readline interface to interact with the agent
+    const rl = readline.createInterface({
+        input: process.stdin,
+        output: process.stdout,
+    });
+
+    while (true) {
+        const prompt = await new Promise<string>((resolve) => {
+            rl.question('Enter your prompt (or "exit" to quit): ', resolve);
+        });
+
+        if (prompt === "exit") {
+            rl.close();
+            break;
+        }
+
+        console.log("\n-------------------\n");
+        console.log("TOOLS CALLED");
+        console.log("\n-------------------\n");
+        try {
+            const result = await generateText({
+                model: openai("gpt-4o-mini"),
+                tools: tools,
+                maxSteps: 10, // Maximum number of tool invocations per request
+                prompt: prompt,
+                onStepFinish: (event) => {
+                    console.log(event.toolResults);
+                },
+            });
+
+            console.log("\n-------------------\n");
+            console.log("RESPONSE");
+            console.log("\n-------------------\n");
+            console.log(result.text);
+        } catch (error) {
+            console.error(error);
+        }
+        console.log("\n-------------------\n");
+    }
+})();
+```
+
+- **Interactive Loop:** This section sets up the command-line interface and the AI interaction loop, identical to previous examples, using `generateText` to process prompts and interact with the configured tools.
+
+### Setup Guide for Celo Token Sending Agent
+
+Follow these steps to set up the AI-powered token sending agent on Celo:
+
+**1. Clone Repository and Navigate to Example Directory:**
+
+Follow steps 1-5 from the NFT minting tutorial (or previous tutorials) to clone the GOAT repository, navigate to the `typescript` directory, install dependencies, build the project, and go to the example directory: `examples/by-use-case/evm-mint-nft`. You can reuse this directory or create a new one if you prefer.
```

Comment:
link the guide to the NFT minting tutorial. mention above that this comes after.